pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -1227,3 +1230,41 @@ type ScalewaySDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// IonosSDConfig configurations allow retrieving scrape targets from Ionos resources.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ionos_sd_config
+type IonosSDConfig struct {
+	// The unique ID of the data center.
+	// +required
+	DatacenterId string `json:"datacenterId"`
+	// Port to scrape the metrics from.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+	// Refresh interval to re-read the list of resources.
+	// +optional
+	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
+	// Optional HTTP basic authentication information, required when using IONOS
```

Comment:
(nit)
```suggestion
	// Optional HTTP basic authentication information, required when using IONOS.
```

Please also add punctuation for the other comments.